java/core/src/main/java/com/whylogs/core/schemas/ColumnSchema.java

```diff
@@ -0,0 +1,24 @@
+package com.whylogs.core.schemas;
+
+import com.whylogs.core.metrics.Metric;
+import com.whylogs.core.metrics.MetricConfig;
+import com.whylogs.core.resolvers.Resolver;
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
+@Data
+@AllArgsConstructor
+public class ColumnSchema {
+  // Thoughts: we could have this ColumnSchema<T> instead of having it as a member
```

Comment:
Here is a thought around passing in the types. Not super important right now as it will become more clear what's easier as I'm testing out the api.